src/main/java/hashtable/HashTable.java

```diff
@@ -62,54 +72,57 @@ public void remove(K key) {
-     * @param key the key whose associated value is to be returned.
-     * @return the value to which the specified key is mapped or null if no value for specified key.
-     * @exception IllegalArgumentException if specified key is null.
+     * Returns the value to which the specified key is mapped
+     * @param key the key whose associated value is to be returned
+     * @return the value to which the specified key is mapped; otherwise the "null" if specified key is null or value with such key does not exist
      */
     public V getValue(K key) {
         if (key == null) {
-            throw new IllegalArgumentException("Key cannot be null!");
+            return null;
         }
         LinkedListNode<K, V> node = getNodeForKey(key);
         return node == null ? null : node.value;
     }
 
     /**
-     * Returns the index in table for specified key.
-     * @param key the key that needs to be converted into index.
-     * @return index in table.
-     * @exception IllegalArgumentException if specified key is null.
+     * @param index slot in a table
+     * @return a string containing all pairs from specified index(slot) or "null" if slot is empty
+     * @throws IllegalArgumentException if specified index is less than 0, equals or more than size of array
      */
-    public int getIndexForKey(K key) {
-        if (key == null) {
-            throw new IllegalArgumentException("Key cannot be null!");
+    String printSlot(int index) {
+        if (index < 0 || index >= array.size()) {
+            throw new IllegalArgumentException("Index cannot be less than 0 or more than size of array!");
         }
-        return Math.abs(key.hashCode() % array.size());
+        String s = array.get(index) == null ? "null" : array.get(index).printContent();
+        return s;
+    }
+
+    int getSize() {
+        return array.size();
     }
 
     /**
-     * Returns a string containing all pairs (key and value) from specified index(slot).
-     * @param index slot in a table
-     * @return a string containing all pairs from specified index(slot) or null if slot is empty.
-     * @exception IllegalArgumentException if specified index is less than 0, equals or more than size of array.
+     * Returns the index in table for specified key
+     * @param key the key that needs to be converted into index
+     * @return index in table
+     * @throws IllegalArgumentException if specified key is null
      */
-    public String printSlot(int index) {
-        if(index < 0 || index >= array.size()) {
-            throw new IllegalArgumentException("Index cannot be less than 0 or more than size of array!");
+    private int getIndexForKey(K key) {
+        if (key == null) {
+            throw new IllegalArgumentException("Key cannot be null!");
         }
-        String s = array.get(index) == null ? "null" : array.get(index).printContent();
-        return s;
+        return Math.abs(key.hashCode() % initialCapacity);
```

Comment:
Again.. it's just `array.length`. I don't think you need `initialCapacity`.

Reply:
I use `initialCapacity` just because arraylist doesn't have `length` method ¯\(°_o)/¯